ignite/metrics/epoch_metric.py

```diff
@@ -133,12 +138,35 @@ def compute(self) -> float:
             # Run compute_fn on zero rank only
             result = self.compute_fn(_prediction_tensor, _target_tensor)
 
+        # compute_fn outputs: scalars, tensors, tuple/list/mapping of tensors.
+        if not _is_scalar_or_collection_of_tensor(result):
+            raise TypeError(
+                "output not supported: compute_fn should return scalar, tensor, tuple/list/mapping of tensors"
+            )
+
         if ws > 1:
             # broadcast result to all processes
-            result = cast(float, idist.broadcast(result, src=0))
+            return apply_to_type(result, torch.Tensor, partial(idist.broadcast, src=0))  # type: ignore[return-value]
```

Comment:
Shouldn't it be
```suggestion
            return apply_to_type(result, (torch.Tensor, int, float), partial(idist.broadcast, src=0))  # type: ignore[return-value]
```

Reply:
Yes, my bad.

I've updated it, but needed to update the ignore type. hopefully now all tests will pass